cmake/external/extensions.cmake

```diff
@@ -11,9 +11,18 @@ set(OCOS_ENABLE_CTEST OFF CACHE INTERNAL "")
 set(OCOS_ENABLE_STATIC_LIB ON CACHE INTERNAL "")
 set(OCOS_ENABLE_SPM_TOKENIZER OFF CACHE INTERNAL "")
 
+# backup CMAKE_CXX_FLAGS in case we'll rewrite it for exceptions flag
+set(CMAKE_CXX_FLAGS_BAK "${CMAKE_CXX_FLAGS}")
+
 # disable exceptions
-if (onnxruntime_DISABLE_EXCEPTIONS)
-  set(OCOS_ENABLE_CPP_EXCEPTIONS OFF CACHE INTERNAL "")
+if(onnxruntime_DISABLE_EXCEPTIONS)
+  # ort-ext needs exceptions enabled for some of the 3rd party libraries.
+  # ort-ext will provide a try/catch layer around all entry points and convert any exceptions to status messages
+  # so no exceptions are passed up to ort
+  set(OCOS_ENABLE_CPP_EXCEPTIONS ON CACHE INTERNAL "")
+  string(REPLACE "-fno-exceptions" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
```

Comment:
Should these changes (-fno-exceptions etc.) be made in onnxruntime-extensions repo, instead of ORT?
And in ort-extensions repo, the fixing should only apply to OpenCV and other libraries which require C++ exceptions.

Reply:
Will try to port to to ort-extensions repo

Reply:
Exceptions will propagate to it's caller, so it means we have to enable exception for the whole extensions, right?